sklearn/metrics/scorer.py

```diff
@@ -66,6 +67,34 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """Check if estimator has a classes argument and if scorer has a
+        labels argument and return new kwargs for scorer if label not already
+        set in make score initialization"""
+        est_has_classes = hasattr(estimator, "classes_")
+        scorer_has_labels = "labels" in signature(
+                                            self._score_func).parameters
+
+        if est_has_classes and scorer_has_labels:
+            classes = getattr(estimator, "classes_", None)
```

Comment:
since scoring works only after fitting, classes_ should have the right classes. it could either be passed using the classes argument or inferred from y. 

i think doing it this way will fix 2 cases- 
1. when test data has subset of classes as train data.
2. when both train and test have subset of classes because partial fit is being used and not all classes are trained now.

if we use "classes" then we'll miss case 1 when user didn't pass the parameter explicitly but train data had all right classes. thoughts?